nixos/modules/services/misc/ollama.nix

```diff
@@ -16,12 +21,22 @@ in {
           Specifies the bind address on which the ollama server HTTP interface listens.
         '';
       };
+      acceleration = lib.mkOption {
```

Comment:
hey, just a small comment. Should this default to config.cudaSupport or config.rocmSupport ?

You can do the same in ollama. Rather than having to make an override here for the ollamaPackage, you can make the build do the right thing if config.cudaSupport is enabled.

As you are the maintainer, I'll let you decide if you like this or not.

Reply:
I'm not sure I understand what you are proposing. Are you saying that the ollama package can take `config` as a parameter? Or are you saying that the `acceleration` option should be split into `rocmSupport` and `cudaSupport`? Could you be more detailed?

Reply:
so inside the ollama package, you could have
```
acceleration ? if config.supportCuda then "cuda" else if config.supportRocm then "rocm" else null
```
then you probably don't need to keep the acceleration option in the module. If people have cuda or rocm enabled, they probably want to use the acceleration automatically and they have the "right" llama package on their system.

Does it make sense ? (let me know if this is still too vague).

Reply:
I mostly understand now, but I'm a little confused about what `config` means when it's in a package vs in a nixos module. At least I'm assuming there's a difference, because when I [searched](https://search.nixos.org/options) for `rocm` and `cuda`, I didn't find anything relevant. I tried looking a little at the nixpkgs source, but I was unable to figure out the difference.

Could you explain how `config` is set for packages and where the value comes from, and how a user would override the default, or point me to any documentation on the topic? I tried looking for documentation, but was unable to find it myself.

Thank you for your help with this.

Reply:
Config is a global nixpkgs setting that is used amongst other things to setup cuda support for a machine.
It can be used in modules or in packages, since its a nixpkgs related setting.
Im not aware of another way to emable cuda on a machine. But it seems you use cuda without this setting , is that correct ?
For an example of where it is used, the torch python package uses it if i remember correctly and llama-cpp for sure.
I am on my phone so its not convenient to put links, but let me know if you need them and ill add them when i get on a computer.
Let me know if anything doesnt make sense.